source/common/formatter/substitution_formatter.h

```diff
@@ -243,6 +244,247 @@ template <class FormatterContext> class FormatterBaseImpl : public FormatterBase
   std::vector<FormatterProviderBasePtr<FormatterContext>> providers_;
 };
 
+// Helper class to write value to output buffer in JSON style.
```

Comment:
Please add a comment here that this class intentiontally duplicates the functionality in Json::Streamer to avoid an impedance mismatch with its API and the usage model required here.

Reply:
Got it.